hw8/todolist/src/components/WeatherWidget/WeatherWidget.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useEffect, useState } from "react";
+import "./WeatherWidget.css";
+
+const defaultLatitude = 41.6938,
+  defaultLongitude = 44.8015;
+
+const prepareData = (): Promise<any> => {
+  return getUserLocation()
+    .then((location: { latitude: number; longitude: number }) => {
+      return getWeatherData(location.latitude, location.longitude);
+    })
+    .catch(() => {
+      return getWeatherData(defaultLatitude, defaultLongitude);
+    });
+};
+
+const WeatherWidget = () => {
+  const [data, setData] = useState<any>(null);
```

Comment:
Don't use any type in your project. You need to create an interface for data type